### PR TITLE
Don't normalize the '-' character in HTTP header names

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Implements https://github.com/open-telemetry/semantic-conventions/pull/369

### CODE REVIEW

**Review by laurit — APPROVED** [review]
Same pattern is used in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/5d6e322f58ab08e61b71d09386e22e28f1048cb4/instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/messaging/CapturedMessageHeadersUtil.java#L33 and https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/5d6e322f58ab08e61b71d09386e22e28f1048cb4/instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletRequestParametersExtractor.java#L80

**Comment by mateuszrzeszutek** [comment]
I'll make the same change for the server parameters feature; the messaging headers will need to be changed anyway when we start working on the messaging semconv.

**Review by trask — APPROVED** [review]

File: instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletRequestParametersExtractor.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ private static AttributeKey<List<String>> createKey(String parameterName) {
     // normalize parameter name similarly as is done with header names when header values are
     // captured as span attributes
     parameterName = parameterName.toLowerCase(Locale.ROOT);
-    String key = "servlet.request.parameter." + parameterName.replace('-', '_');
+    String key = "servlet.request.parameter." + parameterName;
```
Comment (trask) [inline instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletRequestParametersExtractor.java:80]: would it be easy to handle new/old semconv here?

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/CapturedHttpHeadersUtil.java [hunk 2]
```diff
@@ -38,8 +38,8 @@ static AttributeKey<List<String>> responseAttributeKey(String headerName) {
   }
 
   private static AttributeKey<List<String>> createKey(String type, String headerName) {
-    // headerName is always lowercase, see CapturedHttpHeaders
-    String key = "http." + type + ".header." + headerName.replace('-', '_');
+    // headerName is always lowercase, see CapturedHttpHeadersUtil#lowercase
+    String key = "http." + type + ".header." + headerName;
```
Comment (trask) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/CapturedHttpHeadersUtil.java:42]: should have put that last comment here instead ^^

Comment (mateuszrzeszutek) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/CapturedHttpHeadersUtil.java:42]: What comment exactly?

Comment (laurit) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/CapturedHttpHeadersUtil.java:42]: probably
> would it be easy to handle new/old semconv here?

File: instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletRequestParametersExtractor.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletRequestParametersExtractor.java:80]: Done. Not the prettiest, but possible.

**Pull request merged**